site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -113,11 +119,44 @@ def _transform_titles(self):
 
         return rdm_title, rdm_additional_titles
 
+    def _validate_imprint(self):
+        """Validate that record has only 1 imprint."""
+        imprints = self.inspire_metadata.get("imprints", [])
+        if imprints:
+            if len(imprints) > 1:
+                self.metadata_errors.append(
+                    f"More than 1 imprint found. INSPIRE record id: {self.inspire_metadata.get('control_number')}."
+                )
+                return False
+            return imprints[0]
+        return False
+
+    def _transform_publisher(self):
+        """Mapping of publisher."""
+        imprint = self._validate_imprint()
+
+        if imprint:
+            return imprint.get("publisher")
+
+        return None
```

Comment:
```suggestion
        imprint = self._validate_imprint()

        return imprint.get("publisher")
```
the result is the same and we save some lines of code